subprojects/build-cache/src/main/java/org/gradle/caching/internal/origin/OriginMetadataFactory.java

```diff
@@ -43,20 +40,27 @@
     private static final String HOST_NAME_KEY = "hostName";
     private static final String USER_NAME_KEY = "userName";
 
+    private final File rootDir;
     private final String userName;
     private final String operatingSystem;
     private final String currentBuildInvocationId;
-    private final Consumer<Properties> additionalProperties;
-    private final File rootDir;
-
-    private volatile String localHostName;
+    private final PropertiesConfigurator additionalProperties;
+    private final HostnameLookup hostnameLookup;
 
-    public OriginMetadataFactory(File rootDir, String userName, String operatingSystem, String currentBuildInvocationId, Consumer<Properties> additionalProperties) {
+    public OriginMetadataFactory(
+        File rootDir,
+        String userName,
+        String operatingSystem,
+        String currentBuildInvocationId,
+        PropertiesConfigurator additionalProperties,
+        HostnameLookup hostnameLookup
```

Comment:
I can't find the class `HostnameLookup`. Is that missing from the branch?

Reply:
It's down there :)